app/assets/javascripts/notebook/magic/customC3Chart.coffee

```diff
@@ -32,17 +21,15 @@ define([
     console.log("chartOptions", chartOptions)
     chartOptions.bindto = "#" + chart_container.attr("id")
     chartOptions.data = {} if not chartOptions.data
-    chartOptions.data.rows = []
+    chartOptions.data.json = @dataInit
 
-    chart = c3.generate(chartOptions)
+    chartOptions.data.keys = { value: options.headers } if not chartOptions.data.keys or not chartOptions.data.keys.value
 
-    chart.load({ rows: prepare_c3_data(options.headers, @dataInit) })
-    chart.flush()
+    chart = c3.generate(chartOptions)
 
     dataO.subscribe( (newData) =>
-      chart.unload();
-      chart.load({
-        rows: prepare_c3_data(options.headers, newData)
-      });
+      chartOptions.data.json = newData#_.filter(newData, (obj) -> !_.findWhere(currentData, obj))
+      #chartOptions.data.unload = true
```

Comment:
Well indeed! So this is the case where unload is necessary. The problem is that C3 is crappy and sequencing unload with load :-(, they say we have to use the `unload` field in the chart options to do so.

The pitty with unload is that the chart is flicking when you add data because it renders the whole dataset, which is sad and ugly :-(